### PR TITLE
add page on node architecture

### PR DESCRIPTION
**Pull request by jmcook1186**
## Description

Adds a page explaining the architecture of a pos Ethereum node.

## Related Issue

#8285

### CODE REVIEW

File: src/content/developers/docs/nodes-and-clients/node-architecture/index.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+lang: en
+sidebaDepth: 2
+---
+
+An Ethereum node is composed of two clients: an [execution client](https://ethereum.org/en/developers/docs/nodes-and-clients/#execution-clients) and a [consensus client](https://ethereum.org/en/developers/docs/nodes-and-clients/#consensus-clients).
+
+Originally, an execution client alone was enough to run a full Ethereum node. However, ever since Ethereum turned off [proof-of-work](https://ethereum.org/en/developers/docs/consensus-mechanisms/pow/) and implemented [proof-of-stake](https://ethereum.org/en/developers/docs/consensus-mechanisms/pow/), the execution client has needed to be coupled to another piece of software called a [“consensus client”](https://ethereum.org/en/developers/docs/nodes-and-clients/#consensus-clients) in order to keep track of the Ethereum blockchain.
+
+The execution client is responsible for transaction handling, transaction gossip, state management and supporting the Ethereum Virtual Machine ([EVM])(https://ethereum.org/en/developers/docs/evm/). However, it is **not** responsible for block building, block gossiping or handling consensus logic. These are in the remit of the consensus client.
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/node-architecture/index.md:12]:
```suggestion
The execution client is responsible for transaction handling, transaction gossip, state management and supporting the Ethereum Virtual Machine ([EVM](/developers/docs/evm/)). However, it is **not** responsible for block building, block gossiping or handling consensus logic. These are in the remit of the consensus client.
```

**Review by minimalsm — CHANGES_REQUESTED** [review]
Nice :-) Overall looks great 🎉 

The biggest question I have here is about the introduction section. I feel like we might be going a little too deep on execution/consensus explanations before their relevant sections. Quite subjective, but I feel like with it being a relatively complicated subject, it might be worth the effort to consider if we can architect (ha) this page a little more clearly.

File: src/content/developers/docs/nodes-and-clients/node-architecture/index.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+The relationship between the two Ethereum clients is shown in the schematic below. The two clients each connect to their own respective peer-to-peer (P2P) networks. This is because the execution clients gossip transactions over their P2P network enabling them to manage their local transaction pool. The consensus clients gossip blocks over their P2P network, enabling consensus and chain growth.
+
+![node-architecture](node_architecture.png)
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/node-architecture/index.md:16]: Image is illegible in dark mode

File: src/content/developers/docs/nodes-and-clients/node-architecture/index.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+---
+title: Node architecture
+description: Introduction to how Ethereum nodes are organized and where Geth fits.
```
[the remark this answers is not in the document]
Comment (jmcook1186) [reply src/content/developers/docs/nodes-and-clients/node-architecture/index.md:3]: doh, ctrl-c/ctrl-v error

[the remark this answers is not in the document]
Comment (jmcook1186) [reply src/content/developers/docs/nodes-and-clients/node-architecture/index.md:3]:
```suggestion
description: Introduction to how Ethereum nodes are organized.
```

File: src/content/developers/docs/nodes-and-clients/node-architecture/index.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+## Validators {#validators}
+
+Validators can be added to consensus clients if 32 ETH have been sent to the deposit contract. The validator client comes bundled with the consensus client and can be added to a node at any time. The validator handles attestations and block proposals. They enable a node to accrue rewards or lose ETH via penalties or slashing. Running the validator software also makes a node eligible to be selected to propose a new block.
+
+Read more about [proof-of-stake](https://ethereum.org/en/developers/docs/consensus-mechanisms/pos/).
+
```
[the remark this answers is not in the document]
Comment (jmcook1186) [reply src/content/developers/docs/nodes-and-clients/node-architecture/index.md:39]: coming in next commit

File: src/content/developers/docs/nodes-and-clients/node-architecture/index.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+The consensus client does not participate in attesting to or proposing blocks - this is done by a validator, an optional add-on to a consensus client. A consensus client without a validator only keeps up with the head of the chain, allowing the node to stay synced. This enables a user to transact with Ethereum using their execution client, confident that they are on the correct chain.
+
+## Validators {#validators}
+
+Node operators can add a validator to their consensus clients if 32 ETH is the deposit contract. The validator client comes bundled with the consensus client and can be added to a node at any time. The validator handles attestations and block proposals. They enable a node to accrue rewards or lose ETH via penalties or slashing. Running the validator software also makes a node eligible to be selected to propose a new block.
+
+Read more about [proof-of-stake](/developers/docs/consensus-mechanisms/pos/).
+
```
Comment (jmcook1186) [inline src/content/developers/docs/nodes-and-clients/node-architecture/index.md:43]: | Execution Client                                   | Consensus Client                                                 | Validator                    |
| -------------------------------------------------- | ---------------------------------------------------------------- | ---------------------------- |
| Gossips transactions over its p2p network          | Gossips blocks and attestations overs its p2p network            | Proposes blocks              |
| Executes/re-executes transactions                  | Runs the fork choice algorithm                                   | Accrues rewards/penalties    |
| Verifies incoming state changes                    | Keeps track of the head of the chain                             | Makes attestations           |
| Manages state and receipts tries                   | Manages the Beacon state (contains consensus and execution info) | Requires 32 ETH to be staked |
| Creates execution payload                          | Keeps track of accumulated randomness in RANDAO                  | Proposes blocks              |
| Exposes JSON-RPC API for interacting with Ethereum | Keeps track of justification and finalization                    | Can be slashed

Comment (jmcook1186) [inline src/content/developers/docs/nodes-and-clients/node-architecture/index.md:43]:
```suggestion
| Execution Client                                   | Consensus Client                                                 | Validator                    |
| -------------------------------------------------- | ---------------------------------------------------------------- | ---------------------------- |
| Gossips transactions over its p2p network          | Gossips blocks and attestations overs its p2p network            | Proposes blocks              |
| Executes/re-executes transactions                  | Runs the fork choice algorithm                                   | Accrues rewards/penalties    |
| Verifies incoming state changes                    | Keeps track of the head of the chain                             | Makes attestations           |
| Manages state and receipts tries                   | Manages the Beacon state (contains consensus and execution info) | Requires 32 ETH to be staked |
| Creates execution payload                          | Keeps track of accumulated randomness in RANDAO                  | Proposes blocks              |
| Exposes JSON-RPC API for interacting with Ethereum | Keeps track of justification and finalization                    | Can be slashed  
```

File: src/content/developers/docs/nodes-and-clients/node-architecture/index.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+The consensus client does not participate in attesting to or proposing blocks - this is done by a validator, an optional add-on to a consensus client. A consensus client without a validator only keeps up with the head of the chain, allowing the node to stay synced. This enables a user to transact with Ethereum using their execution client, confident that they are on the correct chain.
+
+## Validators {#validators}
+
+Node operators can add a validator to their consensus clients if 32 ETH is the deposit contract. The validator client comes bundled with the consensus client and can be added to a node at any time. The validator handles attestations and block proposals. They enable a node to accrue rewards or lose ETH via penalties or slashing. Running the validator software also makes a node eligible to be selected to propose a new block.
+
+Read more about [proof-of-stake](/developers/docs/consensus-mechanisms/pos/).
```
Comment (jmcook1186) [inline src/content/developers/docs/nodes-and-clients/node-architecture/index.md:42]:
```suggestion
Read more about [staking](/staking).
```

File: src/content/developers/docs/nodes-and-clients/node-architecture/index.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
+---
+title: Node architecture
+description: Introduction to how Ethereum nodes are organized.
+lang: en
+---
+
+An Ethereum node is composed of two clients: an [execution client](https://ethereum.org/en/developers/docs/nodes-and-clients/#execution-clients) and a [consensus client](https://ethereum.org/en/developers/docs/nodes-and-clients/#consensus-clients).
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/node-architecture/index.md:7]:
```suggestion
An Ethereum node is composed of two clients: an [execution client](/developers/docs/nodes-and-clients/#execution-clients) and a [consensus client](/developers/docs/nodes-and-clients/#consensus-clients).
```

File: src/content/developers/docs/nodes-and-clients/node-architecture/index.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+When Ethereum was using [proof-of-work](/developers/docs/consensus-mechanisms/pow/), an execution client was enough to run a full Ethereum node. However, since implementing [proof-of-stake](/developers/docs/consensus-mechanisms/pow/), the execution client needs to be used alongside another piece of software called a [“consensus client”](/developers/docs/nodes-and-clients/#consensus-clients).
+
+The relationship between the two Ethereum clients is shown in the schematic below. The two clients each connect to their own respective peer-to-peer (P2P) networks. This is because the execution clients gossip transactions over their P2P network enabling them to manage their local transaction pool. The consensus clients gossip blocks over their P2P network, enabling consensus and chain growth.
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/node-architecture/index.md:11]:
```suggestion
The diagram below shows the relationship between the two Ethereum clients. The two clients connect to their own respective peer-to-peer (P2P) networks. Separate P2P networks are needed as the execution clients gossip transactions over their P2P network, enabling them to manage their local transaction pool, whilst the consensus clients gossip blocks over their P2P network, enabling consensus and chain growth.
```

File: src/content/developers/docs/nodes-and-clients/node-architecture/index.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+![node-architecture](node-architecture-text-background.png)
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/node-architecture/index.md:13]:
```suggestion
![](node-architecture-text-background.png)
```

File: src/content/developers/docs/nodes-and-clients/node-architecture/index.md [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+| -------------------------------------------------- | ---------------------------------------------------------------- | ---------------------------- |
+| Gossips transactions over its p2p network          | Gossips blocks and attestations overs its p2p network            | Proposes blocks              |
+| Executes/re-executes transactions                  | Runs the fork choice algorithm                                   | Accrues rewards/penalties    |
+| Verifies incoming state changes                    | Keeps track of the head of the chain                             | Makes attestations           |
+| Manages state and receipts tries                   | Manages the Beacon state (contains consensus and execution info) | Requires 32 ETH to be staked |
+| Creates execution payload                          | Keeps track of accumulated randomness in RANDAO                  | Proposes blocks              |
+| Exposes JSON-RPC API for interacting with Ethereum | Keeps track of justification and finalization                    | Can be slashed
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/node-architecture/index.md:50]:
```suggestion
| Exposes JSON-RPC API for interacting with Ethereum | Keeps track of justification and finalization                    | Can be slashed  

```

File: src/content/developers/docs/nodes-and-clients/node-architecture/index.md [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+The consensus client does not participate in attesting to or proposing blocks - this is done by a validator, an optional add-on to a consensus client. A consensus client without a validator only keeps up with the head of the chain, allowing the node to stay synced. This enables a user to transact with Ethereum using their execution client, confident that they are on the correct chain.
+
+## Validators {#validators}
+
+Node operators can add a validator to their consensus clients if 32 ETH is the deposit contract. The validator client comes bundled with the consensus client and can be added to a node at any time. The validator handles attestations and block proposals. They enable a node to accrue rewards or lose ETH via penalties or slashing. Running the validator software also makes a node eligible to be selected to propose a new block.
+
+Read more about [staking](/staking).
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/node-architecture/index.md:42]:
```suggestion
[More on staking](/staking/).

## Components of a node comparison {#node-comparison}


```

**Review by minimalsm — CHANGES_REQUESTED** [review]

File: src/content/developers/docs/nodes-and-clients/node-architecture/index.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+
+## Validators {#validators}
+
+Node operators can add a validator to their consensus clients if 32 ETH is the deposit contract. The validator client comes bundled with the consensus client and can be added to a node at any time. The validator handles attestations and block proposals. They enable a node to accrue rewards or lose ETH via penalties or slashing. Running the validator software also makes a node eligible to be selected to propose a new block.
+
+[More on staking](/staking/).
+
+## Components of a node comparison {#node-comparison}
+
+
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/node-architecture/index.md:46]:
```suggestion
```

File: src/content/developers/docs/nodes-and-clients/node-architecture/index.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+## Validators {#validators}
+
+Node operators can add a validator to their consensus clients if 32 ETH is the deposit contract. The validator client comes bundled with the consensus client and can be added to a node at any time. The validator handles attestations and block proposals. They enable a node to accrue rewards or lose ETH via penalties or slashing. Running the validator software also makes a node eligible to be selected to propose a new block.
+
+[More on staking](/staking/).
+
+## Components of a node comparison {#node-comparison}
+
+| Execution Client                                   | Consensus Client                                                 | Validator                    |
```
Comment (minimalsm) [inline src/content/developers/docs/nodes-and-clients/node-architecture/index.md:46]: The table wasn't rendering correctly in the preview deploy. I think it was due to spacing, if this is fixed this is good to go.

Screenshot 2022-12-09 at 13 23 06

Comment (jmcook1186) [reply src/content/developers/docs/nodes-and-clients/node-architecture/index.md:46]: it's  just missing the final |. Fixing now

**Review by minimalsm — APPROVED** [review]

**Pull request merged**